File: apps/web/src/domain/hypothesis/schemas/dto.ts
import { z } from "zod";
import {
  ENTITY_DESCRIPTION_MAX_LEN,
  ENTITY_NAME_MAX_LEN,
} from "@/lib/validation/entity-limits";

export const insertHypothesisSchema = z.object({
    projectId: z.string().min(1, "Project is required"),
    title: z.string().min(1, "Title is required").max(ENTITY_NAME_MAX_LEN),
    description: z.string().max(ENTITY_DESCRIPTION_MAX_LEN).default(""),
    author: z.string().min(1, "Author is required").max(ENTITY_NAME_MAX_LEN),
    status: z.enum(["proposed", "testing", "supported", "refuted", "inconclusive"]).default("proposed"),
    targetMetrics: z.array(z.string().max(ENTITY_NAME_MAX_LEN)).default([]),
    baseline: z.string().max(ENTITY_NAME_MAX_LEN).default("root"),
  });

File: python/sdk/src/experiment_tracker_sdk/client/domain/hypotheses/dto.py
from datetime import datetime
from enum import Enum
from pydantic import BaseModel


class HypothesisStatus(str, Enum):
    PROPOSED = "proposed"
    TESTING = "testing"
    SUPPORTED = "supported"
    REFUTED = "refuted"
    INCONCLUSIVE = "inconclusive"


class HypothesisCreateRequest(BaseModel):
    projectId: str
    title: str
    description: str = ""
    author: str
    status: HypothesisStatus = HypothesisStatus.PROPOSED
    targetMetrics: list[str] = []
    baseline: str = "root"


class HypothesisUpdateRequest(BaseModel):
    title: str | None = None
    description: str | None = None
    author: str | None = None
    status: HypothesisStatus | None = None
    targetMetrics: list[str] | None = None
    baseline: str | None = None


class HypothesisResponse(BaseModel):
    id: str
    projectId: str
    title: str
    description: str
    author: str
    status: HypothesisStatus
    targetMetrics: list[str]
    baseline: str
    createdAt: datetime
    updatedAt: datetime


class SuccessResponse(BaseModel):
    success: bool
